packages/api/AlvTime.Persistence/Repositories/AccessTokenStorage.cs

```diff
@@ -1,70 +1,63 @@
-﻿using AlvTime.Business.AccessToken;
-using AlvTime.Persistence.DataBaseModels;
+﻿using AlvTime.Persistence.DataBaseModels;
 using System;
 using System.Collections.Generic;
-using System.Globalization;
 using System.Linq;
+using AlvTime.Business.AccessTokens;
+using AlvTime.Business.Interfaces;
+using AlvTime.Business.Models;
 
 namespace AlvTime.Persistence.Repositories
 {
     public class AccessTokenStorage : IAccessTokenStorage
     {
-        private readonly AlvTime_dbContext _context;
+        private readonly AlvTime_dbContext _dbContext;
 
-        public AccessTokenStorage(AlvTime_dbContext context)
+        public AccessTokenStorage(AlvTime_dbContext dbContext)
         {
-            _context = context;
+            _dbContext = dbContext;
         }
 
-        public AccessTokenResponseDto CreateLifetimeToken(string friendlyName, int userId)
+        public AccessTokenDto CreateLifetimeToken(string friendlyName, GenericUser user)
         {
-            var uuid = Guid.NewGuid().ToString();
-
+            var alvUser = _dbContext.User.First(dbUser => dbUser.Email.ToLower().Equals(user.Email.ToLower()));
+            
             var accessToken = new AccessTokens
             {
-                UserId = userId,
-                Value = uuid,
+                UserId = alvUser.Id,
+                Value = Guid.NewGuid().ToString(),
                 ExpiryDate = DateTime.UtcNow.AddMonths(6),
                 FriendlyName = friendlyName
             };
 
-            _context.AccessTokens.Add(accessToken);
-            _context.SaveChanges();
+            _dbContext.AccessTokens.Add(accessToken);
+            _dbContext.SaveChanges();
 
-            return new AccessTokenResponseDto
-            {
-                Token = uuid,
-                ExpiryDate = DateTime.UtcNow.AddMonths(6).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)
-            };
+            return new AccessTokenDto(accessToken.Id, accessToken.Value, accessToken.ExpiryDate, accessToken.FriendlyName);
         }
 
-        public AccessTokenFriendlyNameResponseDto DeleteActiveTokens(int tokenId, int userId)
+        public AccessTokenDto DeleteActiveTokens(int tokenId, GenericUser user)
         {
-            var token = _context.AccessTokens
-                .FirstOrDefault(t => t.Id == tokenId && t.UserId == userId);
+            var alvUser = _dbContext.User.First(dbUser => dbUser.Email.ToLower().Equals(user.Email.ToLower()));
```

Comment:
I would rather let the repository just delete the active user tokens, then in the business I would see if the current user actually is permitted to delete the given input token. It might be that the user only is authenticated to delete it's own tokens, but in theory it should not be too hard to imagine that a user with the correct privileges could delete tokens on behalf of others. Now the "dumb" repository does this kind of logic - which imo is better placed in the business layer